docs/Contributing-back.rst

```diff
@@ -0,0 +1,496 @@
+.. _contributing_back:
+
+Contributing back
+=================
+
+This documentation explains how you can contribute back to EasyBuild,
+and discusses the review process for contributions.
+
+.. contents::
+    :depth: 3
+    :backlinks: none
+
+.. _contributing_back_how:
+
+How to contribute back
+----------------------
+
+.. _contributing_back_feedback:
+
+Giving feedback
+~~~~~~~~~~~~~~~
+
+.. _contributing_back_issues:
+
+Opening issues
+~~~~~~~~~~~~~~
+
+* report problems
+* feature requests
+* change requests
+
+.. _contributing_back_helping_others:
+
+Helping others
+~~~~~~~~~~~~~~
+
+.. _contributing_back_code:
+
+Code contributions
+~~~~~~~~~~~~~~~~~~
+
+.. _contributing_back_easyconfigs:
+
+Easyconfig contributions
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. _contributing_back_docs:
+
+Documentation
+~~~~~~~~~~~~~
+
+
+.. _contributing_back_pull_requests:
+
+Pull requests
+-------------
+
+To contribute back to the EasyBuild framework, easyblocks, easyconfig files
+or the EasyBuild documentation, you will need to open a pull request to the
+corresponding GitHub repository:
+
+* EasyBuild framework: https://github.com/hpcugent/easybuild-framework
+* easyblocks: https://github.com/hpcugent/easybuild-easyblocks
+* easyconfigs: https://github.com/hpcugent/easybuild-easyconfigs
+* documentation: https://github.com/hpcugent/easybuild (see ``docs`` subdirectory)
+
+Doing this the traditional way requires some knowledge about using ``git``
+(i.e. creating commits, using branches, pushing to and pulling from remote
+Git repositories, etc.), and being familiar with GitHub.
+
+However, this can be largely circumvented by using the GitHub integration
+provided by EasyBuild, see :ref:`integration_with_github`.
+
+
+.. _contributing_back_pull_request_setup:
+
+Setting up
+~~~~~~~~~~
+
+.. note:: These instructions assume that you already have a working GitHub account.
+
+          If you don't have a GitHub account yet, visit https://github.com/ to create one.
+
+          Make sure you have registered your SSH public key in your GitHub account,
+          via https://github.com/settings/keys.
```

Comment:
While I agree that this is highly convenient, I think it doesn't belong here. It is not really a requirement, and it might make users wonder why they have to keep introducing their password even with their key uploaded (that happens easily since the default url uses `https` instead of `git+ssh`)